src/sqlfluff/rules/L016.py

```diff
@@ -448,6 +448,12 @@ def _eval(self, segment, raw_stack, **kwargs):
                 else:
                     break
 
+            # Don't even attempt to handle template placeholders as we don't have the content
+            # to move it (appears as empty space to us). It will remain as unfixable.
+            if this_line[-1].type == "placeholder":
```

Comment:
What kind of lines or segments are marked as `placeholder`? I wonder why we don't have access? Maybe because it's not considered part of the text after template expansion?

Reply:
OK that comment isn't entirely accurate so will update it.

`placeholder` are templated areas that don't end up in the final SQL, including Jinja comments (`{#...#}`), assignments (`{% set barry=xxx %}`) or even logic (`{% for barry in pollard %}...{% endfor %}`). If they end up as the cause for long lines then we've limited options anyway.

Yes, in theory, we should be able to fix this:

```sql
SELECT *
FROM {# really long comment #}
```

to this:

```sql
SELECT *
 {# really long comment #}
FROM
```

And similar for `{% set... %}` commands, but we can't fix this:

```sql
{% for... %}
SELECT *
FROM really_long_table_name {% endfor %}
```

And yes the reason we don't have access to the segment, is it's because it's not considered part of the test after template expansion and this rule (mostly) works on the post templated file. We do have access to the source string (in the `.source_str` variable), which is used in the one area we look at pre-templated, but we don't (currently) have access to the original segment. We could add it to handle the above case, but honestly don't think it's worth the complexity.